### PR TITLE
Return step statement from pickles step prover

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR changes the pickles step circuit to handle its public input as a public output instead. This is part of the work towards returning values from circuits, broken out as a PR for ease of review and testing.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

File: src/lib/pickles/step_main.ml [hunk 1]
```diff
@@ -225,6 +225,15 @@ let step_main :
         and prevs =
           exists (Prev_typ.f prev_typs) ~request:(fun () ->
               Req.Proof_with_datas )
+        and unfinalized_proofs =
```
Comment (imeckler) [inline src/lib/pickles/step_main.ml:228]: I guess later we can have the unfinalized proofs get computed right?

File: src/lib/pickles/unfinalized.ml [hunk 2]
```diff
@@ -92,3 +92,12 @@ module Constant = struct
     ; sponge_digest_before_evaluations = Digest.Constant.dummy
     }
 end
+
+let typ ~wrap_rounds : (t, Constant.t) Typ.t =
```
Comment (imeckler) [inline src/lib/pickles/unfinalized.ml:96]: Does the input typ for statement no longer include this stuff?

Comment (mrmr1993) [reply src/lib/pickles/unfinalized.ml:96]: The goal of this PR was to depend on the input `typ` as an 'output' `typ`. To get this data into the circuit, we need to provide a `typ` for it separate from the input/output one.

File: src/lib/pickles/step_main.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/pickles/step_main.ml:228]: Right, the next PR in the chain (#11014) delays this calculation until after we've run the circuit's `main` code.

**Review by imeckler — APPROVED** [review]

**Pull request merged**